libraries/AP_HAL_Linux/RCInput_Aero.cpp

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (C) 2017  Intel Corporation. All rights reserved.
+ *
+ * This file is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+ 
+/* feature added by Anemos Technologies : Philippe Crochat (pcrochat@anemos-technologies.com) */
+
+#include "RCInput_Aero.h"
+#include <stdio.h>
+
+#include <utility>
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_HAL/utility/sparse-endian.h>
+
+using namespace Linux;
+
+#define DEBUG 1
+
+#if DEBUG
+#define debug(fmt, args...) ::printf(fmt "\n", ##args)
+#else
+#define debug(fmt, args...)
+#endif
+
+static const AP_HAL::HAL &hal = AP_HAL::get_HAL();
+
+
+RCInput_Aero::RCInput_Aero()
+{
+	
+}
+
+void RCInput_Aero::init()
+{
+	const uint8_t tx_buffer[2]={CLOCK_REG, 0x00};
+	uint8_t rx_buffer[2];
+	
+	_spi = hal.spi->get_device(DEVICE_NAME);
```

Comment:
you need to get the lock for this device here.

Reply:
we use our custom FPGA firmware. This firmware has been pushed to your FPGA intel github
OK for the lock, we use semaphore right?

Reply:
It would be nice to address the comments there to get it to a state it can be applied.

Reply:
what comments?

Reply:
do you want me to add comments? is that it?

Reply:
lock added, comments added

Reply:
No, I meant that that on the PR to change the FPGA firmware there are lots of unaddressed comments. Some very important ones pointing to buggy behavior. I'm asking the FPGA firmware to be fixed

Reply:
ok, we fixed the SSEL bug that prevented the FPGA to be used alonside the CAN driver. Tell me how we should synchronize both our developments